cmd/create/network/cmd.go

```diff
@@ -0,0 +1,154 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/network"
+	helper "github.com/openshift/rosa/pkg/network"
+	"github.com/openshift/rosa/pkg/ocm"
+	bsOpts "github.com/openshift/rosa/pkg/options/network"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+type NetworkSpec struct {
+	Service network.NetworkService
+}
+
+type networkStruct struct {
+	service network.NetworkService
+}
+
+func NewNetwork(spec NetworkSpec) networkStruct {
+	return networkStruct{
+		service: spec.Service,
+	}
+}
+
+func NewNetworkCommand() *cobra.Command {
+	cmd, options := bsOpts.BuildNetworkCommandWithOptions()
+	cmd.Run = rosa.DefaultRunner(rosa.RuntimeWithOCMAndAWS(), NetworkRunner(options))
+	interactive.AddModeFlag(cmd)
+
+	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
+		templateDir := "cmd/create/network/templates"
+		err := filepath.WalkDir(templateDir, func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if !d.IsDir() && strings.HasSuffix(d.Name(), ".yaml") {
+				templateBody, err := os.ReadFile(path)
+				if err != nil {
+					fmt.Println(err)
+					return nil
+				}
+
+				var templateMap map[string]interface{}
+				err = yaml.Unmarshal(templateBody, &templateMap)
+				if err != nil {
+					fmt.Println(err)
+					return nil
+				}
+
+				parameters, ok := templateMap["Parameters"].(map[string]interface{})
+				if !ok {
+					fmt.Printf("No parameters found in the CloudFormation template %s\n", d.Name())
+					return nil
+				}
+
+				fmt.Printf("Available parameters in %s/%s:\n", filepath.Base(filepath.Dir(path)), d.Name())
+				for paramName := range parameters {
+					fmt.Printf("  %s\n", paramName)
+				}
+			}
+			return nil
+		})
+		if err != nil {
+			fmt.Println(err)
+		}
+
+		fmt.Println("\n" + cmd.UsageString())
+	})
+
+	return cmd
+}
+
+func NetworkRunner(userOptions *bsOpts.NetworkUserOptions) rosa.CommandRunner {
+	return func(ctx context.Context, r *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+		var err error
+		options := NewNetworkOptions()
+		options.args = userOptions
+
+		defer r.Cleanup()
+
+		orgID, _, err := r.OCMClient.GetCurrentOrganization()
+		if err != nil {
+			return r.Reporter.Errorf(err.Error())
+		}
+
+		parsedParams, parsedTags := network.ParseParams(userOptions.Params)
+
+		if parsedParams["Name"] == "" {
+			r.Logger.Debugf("Name not provided, using default name %s", r.Creator.AccountID)
+			parsedParams["Name"] = "rosa-network-stack-" + r.Creator.AccountID
+		}
+		if parsedParams["Region"] == "" {
```

Comment:
if these parsedParams don't exist at all, will these if's error? IIRC go will panic if you try to grab a value from a key which does not exist

Reply:
Only asking because i didnt see a negative test for the validation with the func which parses the params, might be wrong about this either way :+1:

Reply:
I mean considering we will probably be writing these cf templates, the standard for these templates should be that they will at least have a region and a name parameter, otherwise we might get resources scattered across different regions with an unidentifiable resource name. But I'll test this out just to be sure.

Reply:
My main concern is getting a panic rather than a readable error here- no worries if it errors, just to more quickly identify what is missing

Reply:
```
E: failed to create stack, operation error CloudFormation: CreateStack, https response error StatusCode: 400, RequestID: 9ae9fb2e-d8b2-407a-b11c-56dc1e0ad886, api error ValidationError: Parameters: [Region] do not exist in the template
```
This is the error you get if the parameter doesnt exist in the template, it should be fine as is